core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -396,11 +399,53 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
   private[sql] def getHashVar(ctx: CodegenContext,
       keyVars: Seq[String]): Option[String] = getContextObject(ctx, "H", keyVars)
 
-  private[sql] def clearContext(): Unit = synchronized {
-    getContextObject[LogicalPlan](SnappySession.CACHED_PUTINTO_UPDATE_PLAN).
-        foreach { cachedPlan =>
-          sharedState.cacheManager.uncacheQuery(this, cachedPlan, blocking = true)
+  private[sql] def cachePutInto(plan: Option[LogicalPlan], table: String): Boolean = {
```

Comment:
I would not prefer keeping an operation specific logic in SnappySession. We should move it to some other class if possible.